frontend/tracker-web/src/App.js

```diff
@@ -14,44 +17,56 @@ class App extends Component {
     users: [],
     tracks: [],
     currentSection: 'Projects',
+    isAuthenticated: false,
+  }
+
+  fetchData = (endpoint, stateKey) => {
+    fetchWithAuth(`${this.BASE_URL}/api/${endpoint}/`)
+      .then(resp => {
+        if (!resp.ok) throw new Error('Error fetching data');
+        return resp.json();
+      })
+      .then(data => this.setState({ [stateKey]: data }))
+      .catch(error => {
+        console.log(error);
+        this.handleLogout();
+      });
   }
 
   componentDidMount() {
```

Comment:
componentDidMount duplicated with fetch
Try to use functional programming (you used in login file) and hooks instead of class approach in react using useState and etc. aproach. In short - class component outdated style, can be used in hard logic. For easy and fast things we should use function components. Read more https://www.freecodecamp.org/news/function-component-vs-class-component-in-react/